crates/task-impls/src/events.rs

```diff
@@ -183,8 +183,6 @@ pub enum HotShotEvent<TYPES: NodeType> {
     UpgradeVoteSend(UpgradeVote<TYPES>),
     /// Upgrade certificate has been sent to the network
     UpgradeCertificateFormed(UpgradeCertificate<TYPES>),
-    /// A HotShot upgrade was decided
-    UpgradeDecided(UpgradeCertificate<TYPES>),
```

Comment:
Why do we want to remove this?

Reply:
It's an old event; it's not used for anything and shouldn't be used for anything, since the correct way to access version information is via the `upgrade_lock` passed around to the tasks.

(this was just some cleanup, not really related to the PR)